flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -412,31 +418,22 @@ public void createPartition(ObjectPath tablePath, CatalogPartitionSpec partition
 		checkNotNull(partitionSpec);
 		checkNotNull(partition);
 
-		if (!tableExists(tablePath)) {
-			throw new TableNotExistException(catalogName, tablePath);
-		}
-
-		if (!isPartitionedTable(tablePath)) {
-			throw new TableNotPartitionedException(catalogName, tablePath);
-		}
+		ensureTableExists(tablePath);
+		ensurePartitionedTable(tablePath);
+		ensureFullPartitionSpec(tablePath, partitionSpec);
 
 		if (partitionExists(tablePath, partitionSpec)) {
 			if (!ignoreIfExists) {
 				throw new PartitionAlreadyExistsException(catalogName, tablePath, partitionSpec);
 			}
 		}
 
-		if (!isPartitionSpecValid(tablePath, partitionSpec)) {
-			throw new PartitionSpecInvalidException(catalogName, ((CatalogTable) getTable(tablePath)).getPartitionKeys(),
-				tablePath, partitionSpec);
-		}
-
 		partitions.get(tablePath).put(partitionSpec, partition.copy());
 	}
 
 	@Override
 	public void dropPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec, boolean ignoreIfNotExists)
-			throws PartitionNotExistException, CatalogException {
+		throws PartitionNotExistException, CatalogException {
```

Comment:
Nit: Additional tab is needed before throws.

Reply:
Thanks for pointing out, will do